### PR TITLE
Backport #309 and fix assembly

### PR DESCRIPTION
**Pull request by rossabaker**
Quick magit hack while waiting for lunch... brings some of the Scala 3 improvements back to the mainline.

### CODE REVIEW

**Comment by rossabaker** [comment]
I see I clobbered the GraalVM support, which I don't know is in common enough use to be in the main template.

**Review by danicheg — APPROVED** [review]

File: build.sbt [hunk 1]
```diff
@@ -3,39 +3,35 @@
...
 )
 
 val PrimaryOS = "ubuntu-latest"
 val MacOS = "macos-latest"
 ThisBuild / githubWorkflowOSes := Seq(PrimaryOS, MacOS)
+ThisBuild / githubWorkflowJavaVersions := Seq(
+  JavaSpec.temurin("8"),
+  JavaSpec.temurin("11"),
+  JavaSpec.temurin("17")
+)
```
Comment (armanbilge) [inline build.sbt]: If you care enough to bring it back

```suggestion
ThisBuild / githubWorkflowJavaVersions := Seq(
  JavaSpec.temurin("8"),
  JavaSpec.temurin("11"),
  JavaSpec.temurin("17"),
  JavaSpec.graalvm("11")
)
```

Comment (rossabaker) [reply build.sbt]: I'm not sure what's calling assembly then, since that got lost in the merge.

Comment (armanbilge) [reply build.sbt]: Oh, if there are additional CI steps necessary, I don't know about that, sorry!

Comment (rossabaker) [reply build.sbt]: I'll try to keep this as just the backport, but I won't try too hard.

**Comment by rossabaker** [comment]
Whether GraalVM is relevant or not, this failure demonstrates that we prepackage sbt-assembly and then can't assemble it.

**Comment by froth** [comment]
I will have a look at resolving the conflicts that happened in the meantime and get this merged. Hope that's fine.

**Pull request merged**